subscriber/src/service.rs

```diff
@@ -0,0 +1,94 @@
+use auto_farm::common::address_to_id_mapper::{AddressId, NULL_ID};
+
+multiversx_sc::imports!();
+multiversx_sc::derive_imports!();
+
+#[derive(TypeAbi, TopEncode, TopDecode, NestedEncode, NestedDecode, ManagedVecItem, Clone)]
+pub struct PaymentType<M: ManagedTypeApi> {
+    pub opt_specific_token: Option<EgldOrEsdtTokenIdentifier<M>>,
+    pub amount_for_normal: BigUint<M>,
+    pub amount_for_premium: BigUint<M>,
+}
+
+#[derive(TypeAbi, TopEncode, TopDecode, NestedEncode, NestedDecode, ManagedVecItem)]
+pub struct ServiceInfo<M: ManagedTypeApi> {
+    pub sc_address: ManagedAddress<M>,
+    pub payment_type: PaymentType<M>,
+}
+
+#[derive(TypeAbi, TopEncode, TopDecode)]
+pub enum SubscriptionType {
+    None,
+    Daily,
+    Weekly,
+    Monthly,
+}
+
+mod register_service_proxy {
+    #[multiversx_sc::proxy]
+    pub trait RegisterServiceProxy {
+        #[endpoint(registerService)]
+        fn register_service(&self);
+    }
+}
+
+#[multiversx_sc::module]
+pub trait ServiceModule: crate::common_storage::CommonStorageModule {
+    #[only_owner]
+    #[endpoint(registerService)]
+    fn register_service(&self, sc_address: ManagedAddress, payment_type: PaymentType<Self::Api>) {
```

Comment:
you need to specifically register the service. The information of service/payment/periodicity has to be in the subscription contract.

Reply:
This is a must.